esdb-client/src/main/java/com/opencqrs/esdb/client/EventCandidate.java

```diff
@@ -15,8 +15,29 @@
  * @param type uniquely identifies the event type, specifically for being able to interpret the contained data structure
  * @param data a generic map structure containing the event payload, which is going to be stored as JSON within the
  *     event store
+ * @param traceParent the candidate's 'traceparent' header, according to the W3C Trace Context standard
+ * @param traceState the candidate's 'tracestate' header, according to the W3C Trace Context standard
  * @see Event
  * @see EsdbClient#write(List, List)
  */
 public record EventCandidate(
-        @NotBlank String source, @NotBlank String subject, @NotBlank String type, @NotNull Map<String, ?> data) {}
+        @NotBlank String source,
+        @NotBlank String subject,
+        @NotBlank String type,
+        @NotNull Map<String, ?> data,
+        String traceParent,
+        String traceState) {
+
+    /**
+     * Convenience constructor for EventCandidates with no tracing data available.
+     *
+     * @param source
+     * @param subject
+     * @param type
+     * @param data
+     */
+    public EventCandidate(
```

Comment:
please remove. we should always use the constructor with tracing

Reply:
Good Point.